torch/csrc/distributed/c10d/Hooks.cpp

```diff
@@ -0,0 +1,37 @@
+#include <atomic>
+
+#include <deque>
+#include <memory>
+#include <mutex>
+
+#ifndef _WIN32
+#include <unistd.h>
+#else
+#include <io.h>
+#endif
+
+#include <torch/csrc/distributed/c10d/Hooks.hpp>
+namespace c10d {
+
+namespace {
+
+std::mutex callbacks_lock;
+std::vector<CollectiveEventCallback> callbacks_list;
+} // namespace
+
+TORCH_API void register_collective_callback(
+    CollectiveEventCallback&& callback) {
+    TORCH_INTERNAL_ASSERT(0, "NYI, Hooks functionality + tests not landed yet");
+}
+
+namespace details {
+
+void call_collective_callbacks(EventInfo&& evt) {
```

Comment:
Why does this function take as rvalue if all the callbacks are guaranteed to be const ref?